WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.kt

```diff
@@ -669,21 +675,19 @@ class MeFragment : Fragment(R.layout.me_fragment), OnScrollToTopListener {
         }
         val file = File(filePath)
         if (!file.exists()) {
-            ToastUtils.showToast(
-                activity,
-                R.string.error_locating_image,
-                SHORT
-            )
+            ToastUtils.showToast(activity, R.string.error_locating_image, SHORT)
             return
         }
         binding?.showGravatarProgressBar(true)
-        GravatarApi.uploadGravatar(file, accountStore.account.email, accountStore.accessToken,
-            object : GravatarUploadListener {
-                override fun onSuccess() {
+        avatarService.upload(file, Email(accountStore.account.email), accountStore.accessToken!!,
```

Comment:
Maybe we can check if accessToken is not `null` or replace the `!!` with something like `.orEmpty` to prevent the crash in runtime?

Reply:
I'm not sure about doing this:
- accessToken shouldn't be null here (shouldn't be null if the user is logged in).
- if we hide the problem with a `.orEmpty`, we might not detect the problem in a while.

There is one good reason for doing this though:
- Previously the app would not crash here, so we might want to copy the same behavior.

Reply:
> Previously the app would not crash here, so we might want to copy the same behavior.

Yeah, that's my main concern. As the app wouldn't crash before, I wonder if we have some cases where `accessToken` is `null.` (I agree it shouldn't).

I've seen we have `AppLog`. I'm not sure how it works, but maybe we can track that error.

Reply:
@antonis any opinion on that one? Do you remember a similar situation elsewhere?

Reply:
I agree that having a `null` access token is an edge case but I think it's a good idea to [add a check as in other cases](https://github.com/search?q=repo%3Awordpress-mobile%2FWordPress-Android+accountStore.accessToken.isNullOrEmpty%28%29&type=code) given that the `AccountStore.getAccessToken` method returns a `@Nullable` value.

Reply:
Ok, I added the `.orEmpty` check in [1bdd94d](https://github.com/wordpress-mobile/WordPress-Android/pull/20590/commits/1bdd94d3356d2fe34e71f7f8f4d56b0399977594).

If the token is null, the endpoint will return an "Invalid Token" error that will be logged under the track event `ME_GRAVATAR_UPLOAD_EXCEPTION` with a property `error_type` set to `invalid_token`. We'll also show an error toast like for any upload error via the message `GravatarUploadFinished` we send through the event bus.